portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,10 +37,23 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<Comment> comments = new ArrayList<Comment>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()){
+      String message = (String) entity.getProperty("comment");
```

Comment:
Totally optional topic you can research if you're feeling like an extra challenge: I think Datastore might have a limit for how long a String can be? What happens if you try to post a really long comment?

(Like I said, this is totally optional! Feel free to leave this for later or to disregard it entirely.)